divide_and_conquer/closest_pair_of_points.py

```diff
@@ -0,0 +1,67 @@
+"""
+The algorithm finds distance between closest pair of points in the given n points.
+Approach used -> Divide and conquer 
+The points are sorted based on x-cords 
+& by applying divide and conquer approach, 
+minimum distance is obtained recursively.
+
+Edge case: closest points lie on different sides of partition
+This case handled by forming a strip of points 
+which are at a distance (< closest_pair_dis) from mid-point.
+(It is a proven that strip contains at most 6 points)
+And brute force method is applied on strip to find closest points. 
+
+Time complexity: O(n * (logn) ^ 2)
+"""
+
+
+import math 
+
+
+def euclidean_distance(point1, point2):
+    return math.sqrt(pow(point1[0] - point2[0], 2) + pow(point1[1] - point2[1], 2))
```

Comment:
It's OK but actually sqrt is unnecessary. For the optimization, you can use sqrt only once after all calculations have been done.

Reply:
ok.